app/Console/Commands/AdserversCrawlCommand.php

```diff
@@ -0,0 +1,180 @@
+            ];
+            var_dump($logs);
+            // currently it enables to process single own host without blockchain
+//             throw $e;
+        }
+
+        // TODO: check with Jacek - smell logic bugs
+        // TODO: review adseelct json generation filters and double check if this could be copied into adserver database
+
+        if ($logs) {
+            foreach ($logs as $log) {
+                $host = hex2bin($log['message']);
+                if ($this->host == $host) {
+                    // found own broadcast. No need to send it again
+                    $this->broadcast = false;
+                }
+                $this->info("Found $host -> {$log['address']}");
+                // TODO: extract algo
+                if (preg_match('/^([a-z0-9][a-z0-9-]{0,62}\.)+([a-z]{2,})$/i', $host)) {
+                    NetworkHost::registerHost(Esc::normalizeAddress($log['address']), $host);
+                // TODO: check this with Jacek in adserver symfony code
+                    // $nHost->setAccountMsid($log['account_msid']);
+                } else {
+                    // TODO: debug error log?
+                }
+            }
+        }
+
+        if ($this->broadcast) {
+            $this->info("Broadcast own host: $this->host");
+            $x = $this->esc->sendBroadcast($this->host);
+            var_dump($x);
+            // TODO: this fails currently
+//             die(print_r($x));
+        }
+    }
+
+    protected function crawlHosts()
+    {
+        $crawlTime = time();
+
+        $hosts = NetworkHost::where('last_seen', '>', time() - $this->crawlHostsIfLastSeenLimit)->get();
+
+        $batch = 0;
+
+        // TODO: integrate with adselect
+        // $adselectService = $this->getContainer()->has('adselect') ? $this->getContainer()->get('adselect') : null;
+        // $adselectService instanceof Adselect;
+        // $adselectCmp = [];
+
+        foreach ($hosts as $r) {
+            $host = $r->host;
+            $this->info("Crawl: $host");
+
+            // status: updated, removed, synced (adselect)
+            $inventory = json_decode(file_get_contents("http://{$host}/adshares/inventory/list"), JSON_OBJECT_AS_ARRAY);
+
+            if (empty($inventory)) {
+                // TODO: double check empty behaviour
+                continue;
+            }
+
+            foreach ($inventory['campaigns'] as $campaign_data) {
+                $campaign_data['source_host'] = $host;
+                $campaign = NetworkCampaign::fromJsonData($campaign_data);//, $existing);
+
+                // $adselectCmp[] = $campaign->getAdselectJson();
+                // if ($batch++ == 100) {
+                //     if ($adselectService) {
+                //         $adselectService->addCampaigns($adselectCmp);
+                //         $adselectCmp = [];
+                //     }
+                //     $em->flush();
+                // }
+            }
+
+            // if ($adselectService) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS125)